.github/dotnet.sh

```diff
@@ -3,4 +3,4 @@
 curl -sSL https://packages.microsoft.com/keys/microsoft.asc | sudo tee /etc/apt/trusted.gpg.d/microsoft.asc
 sudo apt-add-repository https://packages.microsoft.com/ubuntu/22.04/prod
 sudo apt-get update
-sudo apt-get install -y libmsquic zlib1g-dev
+sudo apt-get install -y libmsquic zlib1g-dev clang
```

Comment:
haven't validated this is sufficient to get nativeaot to build in the container. if necessary we can remove it from bench.yml for now as the other changes should be fine.